lang/java/reef-utils/src/main/java/org/apache/reef/util/AsynchronousCaller.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.util;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.concurrent.Callable;
+
+/**
+ * Generic class which provides a simple mechanism to call a single method
+ * on a concrete class.
+ * @param <TReturn> The class type of the method return value.
+ * @param <TInput> The class type of the method input parameter.
+ * @param <TClass> The class type of the class on which the method resides.
+ */
+public final class AsynchronousCaller<
+      TReturn extends Object, TInput extends Object, TClass extends Object> implements Callable<TReturn> {
+  private TClass obj;
+  private TInput input;
+  private Method method;
```

Comment:
I already tried and I was getting  `a variable xxx might not have been initialized error`. However, it was because I had declared the default constructor to be private. Works now.
Done